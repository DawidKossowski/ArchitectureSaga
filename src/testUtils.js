import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const initialReducerState = {
    users: [],
};

export function wrapInProvider(elementToWrap) {
    const mockStore = configureStore();
    const store = mockStore(initialReducerState);

    return (
      <Provider store={store}>
          {elementToWrap}
      </Provider>
    );
}
