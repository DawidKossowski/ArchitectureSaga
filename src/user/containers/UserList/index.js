import React, {Component} from 'react';
import {connect} from 'react-redux';
import {fetchUsers} from '../../userActions';
import User from "../../components/User";

class UserList extends Component {
    componentDidMount() {
        this.props.fetchUsers();
    }

    render() {
        const {users} = this.props;
        return (
          <ul>
              {users.map(user => (
                <User data={user} key={user.id}/>
              ))}
          </ul>
        )
    }
}

const mapStateToProps = state => ({
    users: state.users
});

const actions = {
    fetchUsers
};

export default connect(mapStateToProps, actions)(UserList);
