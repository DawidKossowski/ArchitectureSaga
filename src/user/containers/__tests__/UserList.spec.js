import React from 'react';
import renderer from 'react-test-renderer'
import UserList from '../UserList';
import { wrapInProvider } from '../../../testUtils';

describe('user sagas', () => {
    const props = {
        users: [
            { id: 1, first_name: 'D', last_name: 'K'}
        ],
    };
    it('should fetch users', () => {
        const user = renderer.create(
          wrapInProvider(<UserList />)
        ).toJSON();
        expect(user).toMatchSnapshot();
    });
});
