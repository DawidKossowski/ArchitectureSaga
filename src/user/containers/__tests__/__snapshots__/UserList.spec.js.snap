// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user sagas should fetch users 1`] = `<ul />`;
